mindone/utils/amp.py

```diff
@@ -0,0 +1,59 @@
+import mindspore as ms
+from mindspore import nn
+from mindspore.train.amp import AMP_BLACK_LIST, AMP_WHITE_LIST, _auto_black_list, _auto_white_list
+
+
+def auto_mixed_precision(network, amp_level="O0"):
```

Comment:
It's not good to expose internal functions (those starting with `_`) as they may change in the future without backward compatibility (as we can see here with `_auto_black_list` which requires `try-except` wrapping). A better approach would be:
```python
    if amp_level != "O2":
        network = amp.auto_mixed_precision(network, amp_level)
    else:
        network = amp.custom_mixed_precision(network, black_list=AMP_BLACK_LIST + [nn.GroupNorm])

    return network
```
I would also make it possible to add additional layers to the exception list through the function's parameters.

Reply:
Good advance. It takes time to refractor this part. Maybe in next PR